src/test/jtx/TrustedPublisherServer.h

```diff
@@ -94,22 +186,41 @@ class TrustedPublisherServer
             return l.str();
         };
 
-        acceptor_.open(ep.protocol());
+        if (useSSL_)
+        {
+            // This holds the self-signed certificate used by the server
+            load_server_certificate();
+        }
+
+        if (immediateStart)
+            start();
+    }
+
+    void start()
+    {
         error_code ec;
+        acceptor_.open(ep_.protocol());
         acceptor_.set_option(
             boost::asio::ip::tcp::acceptor::reuse_address(true), ec);
-        acceptor_.bind(ep);
+        acceptor_.bind(ep_);
         acceptor_.listen(boost::asio::socket_base::max_connections);
         acceptor_.async_accept(
             sock_,
             std::bind(
                 &TrustedPublisherServer::on_accept, this, std::placeholders::_1));
     }
 
-    ~TrustedPublisherServer()
+    void stop()
     {
         error_code ec;
         acceptor_.close(ec);
+        // TODO consider making this join
+        // any running do_peer threads
```

Comment:
Probably a good idea, or maintain a reference count and have the dtor block.

Reply:
I wanted to do this, but I think we'd have to convert to async calls. That's not horrible, but more work than I want to do here. I'll make a low priority issue